src/Network.js

```diff
@@ -39,180 +36,110 @@ function build_graph(trustlineUpdateEvents) {
-          }
-          const eventMap = filterEvents(events, {
-            TrustlineUpdate: "trustlineUpdateEvents",
-            Transfer: "transferEvents",
-          })
-          const [nodes, edges] = build_graph(eventMap.trustlineUpdateEvents)
-
-          if (nodeSet == null) {
-            console.log("Init network data")
-            nodeSet = new vis.DataSet(nodes)
-            edgeSet = new vis.DataSet(edges)
-            const data = {
-              nodes: nodeSet,
-              edges: edgeSet,
-            }
-            visNetwork.setData(data)
-            visNetwork.once("stabilizationIterationsDone", params => {
-              finishedLoading = true
-              setLoadingPercent(100)
-            })
-          } else {
-            console.log("Update network data")
-            nodeSet.update(nodes)
-            edgeSet.update(edges)
-            finishedLoading = true
-          }
-
-          setNumTransfers(
-            numTransfers => numTransfers + eventMap.transferEvents.length
-          )
-          setNumUsers(nodeSet.length)
-          setLastUpdateTimestamp(Date.now())
-          fromBlock = events[events.length - 1].blockNumber + 1
+        const eventMap = filterEvents(events, {
+          TrustlineUpdate: "trustlineUpdateEvents",
+          Transfer: "transferEvents",
+        })
+        const [nodes, edges] = build_graph(eventMap.trustlineUpdateEvents)
+
+        const newGraph = {
+          links: [...networkGraph.links, ...edges],
+          nodes: [...networkGraph.nodes, ...nodes],
         }
-
-        _fetchMoreData()
-        return setInterval(_fetchMoreData, UPDATE_INTERVAL)
+        networkGraph = newGraph
+        finishedLoading = true
+        setGraph(newGraph)
+
+        setNumTransfers(
+          (numTransfers) => numTransfers + eventMap.transferEvents.length
+        )
+        setNumUsers(networkGraph.nodes.length)
+        setLastUpdateTimestamp(Date.now())
+        fromBlock = events[events.length - 1].blockNumber + 1
+        setLoadingPercent(100)
       }
 
-      const intervalId = startFetchData()
+      return _fetchMoreData()
+    }
 
-      return () => clearInterval(intervalId)
-    },
-    [visNetwork, network]
-  )
+    startFetchData()
+  }
+
+  useEffect(fetchData, [network])
+
+  useEffect(() => {
+    chart(graph, network.address, onSelectTrustline, onSelectAccount)
+  }, [graph, network, onSelectTrustline, onSelectAccount])
 
   return (
-    <div style={{ width: "100%", height: "100%" }}>
-      <NetworkInfo
-        networkStatic={network}
-        numUsers={numUsers}
-        numTransfers={numTransfers}
-      />
-      last updated: <DiffTime timestamp={lastUpdateTimestamp} />
-      {loadingPercent !== 100 && (
-        <progress
-          className="progress my-progress is-info"
-          value={loadingPercent}
-          max="100"
-        >
-          {loadingPercent}%
-        </progress>
+    <div>
+      {loadingPercent !== 100 ? (
+        "Loading..."
```

Comment:
Any specific reason for hiding the `<NetworkInfo />` before everything is loaded? When changing between networks is looks a bit jumpy IMO.

Reply:
when changing the network info shows 0 for users and transfers - which is confusing. That's why I opted for hiding. Better would be to display the network name and display loading below it, but yea... I've now set minHeight of 100px so that keeps the graph in the same spot.